predicate/src/rpc_predicate.rs

```
@@ -395,56 +379,97 @@ mod tests {
-        let mut rewriter = FieldColumnRewriter {
-            field_projections: &mut field_columns,
-        };
+    fn test_normalize_predicate_field_rewrite() {
+        let predicate = normalize_predicate(
+            "table",
+            Some(schema()),
+            &PredicateBuilder::new()
+                .add_expr(col("_field").eq(lit("f1")))
+                .build(),
+        )
+        .unwrap();
+
+        let expected = PredicateBuilder::new()
+            .field_columns(vec!["f1"])
+            .add_expr(lit(true))
+            .build();
+
+        assert_eq!(predicate, expected);
+    }
 
-        let cases = vec![
-            (
-                binary_expr(col("f1"), Operator::Eq, lit(1.82)),
-                binary_expr(col("f1"), Operator::Eq, lit(1.82)),
-                vec![],
-            ),
-            (
-                // TODO - should be rewritten and project onto all field columns
-                binary_expr(col(FIELD_COLUMN_NAME), Operator::NotEq, lit("foo")),
-                binary_expr(col(FIELD_COLUMN_NAME), Operator::NotEq, lit("foo")),
-                vec![],
-            ),
-            (
-                binary_expr(col(FIELD_COLUMN_NAME), Operator::Eq, lit("f1")),
-                lit(true),
-                vec!["f1"],
-            ),
-            (
-                binary_expr(
-                    binary_expr(col(FIELD_COLUMN_NAME), Operator::Eq, lit("f1")),
-                    Operator::Or,
-                    binary_expr(col(FIELD_COLUMN_NAME), Operator::Eq, lit("f2")),
-                ),
-                binary_expr(lit(true), Operator::Or, lit(true)),
-                vec!["f1", "f2"],
-            ),
-        ];
+    #[test]
+    fn test_normalize_predicate_field_rewrite_multi_field() {
+        let predicate = normalize_predicate(
+            "table",
+            Some(schema()),
+            &PredicateBuilder::new()
+                .add_expr(col("_field").eq(lit("f1")).or(col("_field").eq(lit("f2"))))
```

What happens if these to EQs are linked via AND instead of OR?

---

This is a tricky question. I guess similar question to `_field != col_a OR _field != col_b`. Should we throw syntax error?

---

> What happens if these to EQs are linked via AND instead of OR?

Yes, I agree  it should throw an error (which is also covered in the field_rewrite module) but I will add a test demonstrating that here as well.